cellprofiler/settings.py

```diff
@@ -722,6 +722,25 @@ def set_value_text(self, value_text):
             self.__default = self.str_to_value(value_text)
         except:
             logger.debug("Number set to illegal value: %s" % value_text)
+            
+    def set_min_value(self, minval):
+        '''Programatically set the minimum value allowed'''
+        self.__minval = minval
+
+    def set_max_value(self, minval):
+        '''Programatically set the maximum value allowed'''
+        self.__maxval = maxval
+        
+    def get_min_value(self):
+        '''The minimum value (inclusive) that can legally be entered'''
+        return self.__minval
+    
+    def get_max_value(self):
+        '''The maximum value (inclusive) that can legally be entered'''
+        return self.__maxval
+    
```

Comment:
The accessors are added with the property() function below. You can see above how `self.object_size.min_value` is referenced. If you want the separate decorators for stylistic reasons, feel free to change.